### ISSUE TITLE
Time#usec may be negative

### ISSUE THREAD

**Issue by kou** [mruby/mruby#6059]
I don't know why but `(Time.now - (0.1 * XXX)).usec` may be negative:

```console
$ build/host/bin/mruby -v -e 'p((Time.now - (0.1 * 60)).usec)'
mruby 3.2.0 (2023-02-24)
00001 NODE_SCOPE:
00001   NODE_BEGIN:
00001     NODE_FCALL:
00001       method='p' (27)
00001       args:
00001         NODE_CALL(.):
00001           NODE_BEGIN:
00001             NODE_CALL(.):
00001               NODE_CALL(.):
00001                 NODE_CONST Time
00001                 method='now' (132)
00001               method='-' (6)
00001               args:
00001                 NODE_BEGIN:
00001                   NODE_CALL(.):
00001                     NODE_FLOAT 0.1
00001                     method='*' (4)
00001                     args:
00001                       NODE_INT 60 base 10
00001           method='usec' (334)
irep 0x55ac57d79840 nregs=6 nlocals=1 pools=1 syms=4 reps=0 ilen=28
file: -e
    1 000 GETCONST	R2	Time	
    1 003 SEND		R2	:now	n=0
    1 007 LOADL		R3	L[0]	; 0.100000
    1 010 LOADI		R4	60	
    1 013 MUL		R3	R4
    1 015 SUB		R2	R3
    1 017 SEND		R2	:usec	n=0
    1 021 SSEND		R1	:p	n=1
    1 025 RETURN	R1		
    1 027 STOP

-995555
```

Note: If we don't use `Float#*` (`(Time.now - 0.1).usec`), `usec` isn't negative:

```console
$  build/host/bin/mruby -v -e 'p((Time.now - 0.1).usec)'
mruby 3.2.0 (2023-02-24)
00001 NODE_SCOPE:
00001   NODE_BEGIN:
00001     NODE_FCALL:
00001       method='p' (27)
00001       args:
00001         NODE_CALL(.):
00001           NODE_BEGIN:
00001             NODE_CALL(.):
00001               NODE_CALL(.):
00001                 NODE_CONST Time
00001                 method='now' (132)
00001               method='-' (6)
00001               args:
00001                 NODE_FLOAT 0.1
00001           method='usec' (334)
irep 0x561dcd8e6840 nregs=5 nlocals=1 pools=1 syms=4 reps=0 ilen=23
file: -e
    1 000 GETCONST	R2	Time	
    1 003 SEND		R2	:now	n=0
    1 007 LOADL		R3	L[0]	; 0.100000
    1 010 SUB		R2	R3
    1 012 SEND		R2	:usec	n=0
    1 016 SSEND		R1	:p	n=1
    1 020 RETURN	R1		
    1 022 STOP

728826
```

**Comment by kou** [comment]
One more example with `0.1 * 1`:

```console
$ build/host/bin/mruby -v -e 'p((Time.now - (0.1 * 1)).usec)'
mruby 3.2.0 (2023-02-24)
00001 NODE_SCOPE:
00001   NODE_BEGIN:
00001     NODE_FCALL:
00001       method='p' (27)
00001       args:
00001         NODE_CALL(.):
00001           NODE_BEGIN:
00001             NODE_CALL(.):
00001               NODE_CALL(.):
00001                 NODE_CONST Time
00001                 method='now' (132)
00001               method='-' (6)
00001               args:
00001                 NODE_BEGIN:
00001                   NODE_CALL(.):
00001                     NODE_FLOAT 0.1
00001                     method='*' (4)
00001                     args:
00001                       NODE_INT 1 base 10
00001           method='usec' (334)
irep 0x55e5e1788840 nregs=6 nlocals=1 pools=1 syms=4 reps=0 ilen=27
file: -e
    1 000 GETCONST	R2	Time	
    1 003 SEND		R2	:now	n=0
    1 007 LOADL		R3	L[0]	; 0.100000
    1 010 LOADI_1	R4	(1)	
    1 012 MUL		R3	R4
    1 014 SUB		R2	R3
    1 016 SEND		R2	:usec	n=0
    1 020 SSEND		R1	:p	n=1
    1 024 RETURN	R1		
    1 026 STOP

-4127
```

**Comment by matz** [comment]
It's a bug in dc88a87f

**Comment by matz** [comment]
Fixed by b064d7e547
